File: src/extension/compass/__tests__/fixtures/type_hints.go
package demo

type OrganizationService struct {
}

type Logger struct {
}

type Repository struct {
}

type TenantController struct {
	logger *Logger
}

func NewController(service *OrganizationService) *TenantController {
	return &TenantController{}
}

func (c *TenantController) Resolve(repo Repository) *OrganizationService {
	return nil
}

func (c *TenantController) Count(value int) string {
	return ""
}

File: src/extension/compass/__tests__/fixtures/type_hints.php
<?php

namespace App\Services;

class OrganizationService
{
}

class Logger
{
}

class Repository
{
}

class TenantController
{
    private Logger $logger;

    public function __construct(private OrganizationService $service)
    {
    }

    public function resolve(Repository $repo): OrganizationService
    {
        return $this->service;
    }

    public function count(int $value): string
    {
        return (string) $value;
    }
}
